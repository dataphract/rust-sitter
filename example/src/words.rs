#[rust_sitter::grammar("words")]
pub mod grammar {
    #[rust_sitter::language]
    #[derive(Debug)]
    pub struct Words {
        #[rust_sitter::repeat]
        _words: Vec<Word>,
    }

    #[derive(Debug)]
    enum Word {
        Keyword(
            #[rust_sitter::leaf(text = r"if")]
            #[rust_sitter::word]
            (),
        ),
        Ident(
            #[rust_sitter::leaf(pattern = r"[a-zA-Z_][a-zA-Z0-9_]*", transform = |v| v.to_string())]
             String,
        ),
    }

    #[rust_sitter::extra]
    struct Whitespace {
        #[rust_sitter::leaf(pattern = r"\s")]
        _whitespace: (),
    }
}

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn words_grammar() {
        insta::assert_debug_snapshot!(grammar::parse("if"));
        insta::assert_debug_snapshot!(grammar::parse("hello"));
        insta::assert_debug_snapshot!(grammar::parse("ifhello"));
        insta::assert_debug_snapshot!(grammar::parse("if hello"));
    }
}
