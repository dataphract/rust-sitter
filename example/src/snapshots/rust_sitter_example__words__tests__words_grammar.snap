---
source: example/src/words.rs
expression: "grammar::parse(\"if\")"
---
Err(
    [
        ParseError {
            reason: FailedNode(
                [],
            ),
            start: 0,
            end: 2,
        },
    ],
)
